packages/providers/src/provider.ts

```diff
@@ -235,12 +245,17 @@ export default class Provider {
    * Submits a transaction to the chain to be executed
    */
   async sendTransaction(
-    transactionRequestLike: TransactionRequestLike
+    transactionRequestLike: TransactionRequestLike,
+    { signTransaction }: Partial<ProviderCallParams> = {}
```

Comment:
I have some thoughts on this;
1. The signing process should take place before the tx arrives at the provider level.
2. `getReceiptsWithMissingOutputVariables` should always run on dryRun mode with utxValidation: off, avoiding signing a tx on its try. Also, imagining a scenario where the user needs to approve every try would make the UX bad.